src/components/Editor/PostEditPage.js

```diff
@@ -0,0 +1,108 @@
+import { EditorPages, EditorPagesFooter } from './index.js';
+import { getItem, removeItem, setItem } from '../../../utils/storage.js';
+import { request } from '../../api/api.js';
+
+
+export default function PostEditPage({$target, initialState, onMenuCorrection}) {
+    const $page = document.createElement('div');
+    $page.classList.add('editor-wrap')
+
+
+    this.state = initialState;
+
+    let postLocalSaveKey = `post-temp-${this.state.id}`;
+    const post = getItem(postLocalSaveKey, {
+        title: '',
+        content: ''
+    })
+
+    let timer = null;
+
+    const editor = new EditorPages({
+        $target: $page,
+        initialState: post,
+        onEditing: (post) => {
+            if(timer !== null) {
+                clearTimeout(timer);
+            }
+            timer = setTimeout(async () => {
+                setItem(postLocalSaveKey, {
+                    ...post,
+                    tempSaveDate: new Date()
+                })
+
+                if(this.state.id === 'new') {
+                    const createdPosts = await request('/documents', {
+                        method: 'POST',
+                        body: JSON.stringify(post),
+                    })
+
+                    history.replaceState(null, null, `/documents/${createdPosts.id}`);
+                    removeItem(postLocalSaveKey);
+                } else {
+                    await request(`/documents/${post.id}`, {
+                        method: 'PUT',
+                        body: JSON.stringify(post),
+                    });
+                    removeItem(postLocalSaveKey);
+                }
+
+                onMenuCorrection();
+            }, 1000);
+        }
+    });
+
+    this.setState = async (newState) => {
+        if(this.state.id !== newState.id) {
+            this.state = newState;
+            postLocalSaveKey = `post-temp-${this.state.id}`;
+            await fetchPost();
+        }
+
+        this.state = newState;
+        
+        this.render();
+
+        if(this.state.post) {
+            editor.setState(this.state.post);
+            editorFooter.setState(this.state.post);
+        }
+    }

```

Comment:
(제가 구조를 잘 못 이해하고 말씀드리는 걸 수도 있습니다 ㅠㅠ!!)

일단은 this.state = newState 가 중복되니 둘 중 하나는 없어도 괜찮을 것 같습니다 ㅎㅎ

그리고 id는 다를 때만 동작하고, post는 그냥 존재하기만 하면 setState로 하위 컴포넌트 상태도 같이 업데이트 해주고 있는데, 통일성이 있으면 좋을 것 같아요!

```suggestion
    this.setState = async (newState) => {
        this.state = newState;
        editor.setState(this.state.post);
        editorFooter.setState(this.state.post);
            
        postLocalSaveKey = `post-temp-${this.state.id}`;
        await fetchPost();
    }

```

그냥 이렇게 해도 괜찮지않을까요? 어차피 보통 setState는 무언가 변화한 상황에서 부르게 되어있으니깐요 ㅎㅎ 꼭 id가 바뀌었을 때라던가 따지지 않아도 성능 면에서 나쁘지 않을 것 같습니다 ~!